code/modules/mob/living/carbon/human/life.dm

```diff
@@ -331,13 +334,187 @@
+		// Too much exhaled gas in the air
+		var/word
+		var/warn_prob
+		var/oxyloss
+		var/alert
+		if(exhaled_pp > safe_exhaled_max)
+			word = pick("extremely dizzy","short of breath","faint","confused")
+			warn_prob = 15
+			oxyloss = HUMAN_MAX_OXYLOSS
+			alert = 1
+			failed_exhale = 1
+		else if(exhaled_pp > safe_exhaled_max * 0.7)
+			word = pick("dizzy","short of breath","faint","momentarily confused")
+			warn_prob = 1
+			alert = 1
+			failed_exhale = 1
+			var/ratio = 1 - (safe_exhaled_max - exhaled_pp)/(safe_exhaled_max*0.3)
+			if (getOxyLoss() < 50*ratio)
+				oxyloss = HUMAN_MAX_OXYLOSS
+		else if(exhaled_pp > safe_exhaled_max * 0.6)
+			word = pick("a little dizzy","short of breath")
+			warn_prob = 1
+		else
+			co2_alert = 0
+
+		if(!co2_alert && word && prob(warn_prob))
+			to_chat(src, SPAN_WARNING("You feel [word]."))
+			adjustOxyLoss(oxyloss)
+			co2_alert = alert
+
+	// Too much poison in the air.
+
+	if(toxins_pp > safe_toxins_max)
+		var/ratio = (poison/safe_toxins_max) * 10
+		reagents.add_reagent("toxin", CLAMP(ratio, MIN_TOXIN_DAMAGE, MAX_TOXIN_DAMAGE))
+		breath.adjust_gas(poison_type, -poison/6, update = 0) //update after
+		plasma_alert = 1
+	else
+		plasma_alert = 0
+
+	// If there's some other shit in the air lets deal with it here.
+	if(breath.gas["sleeping_agent"])
+		var/SA_pp = (breath.gas["sleeping_agent"] / breath.total_moles) * breath_pressure
+		if(SA_pp > SA_para_min)		// Enough to make us paralysed for a bit
+			Paralyse(3)	// 3 gives them one second to wake up and run away a bit!
+			if(SA_pp > SA_sleep_min)	// Enough to make us sleep as well
+				Sleeping(5)
+		else if(SA_pp > 0.15)	// There is sleeping gas in their lungs, but only a little, so give them a bit of a warning
+			if(prob(20))
+				emote(pick("giggle", "laugh"))
+
+		breath.adjust_gas("sleeping_agent", -breath.gas["sleeping_agent"]/6, update = 0) //update after
+
+	// Were we able to breathe?
+	var/failed_breath = failed_inhale || failed_exhale
+	if (!failed_breath)
+		adjustOxyLoss(-5)
+
+	handle_temperature_effects(breath)
+
+	breath.update_values()
+	return !failed_breath
+
+/mob/living/carbon/human/proc/handle_temperature_effects(datum/gas_mixture/breath)
+	if(!species)
+		return
+	// Hot air hurts :( :(
+	if((breath.temperature < species.cold_level_1 || breath.temperature > species.heat_level_1) && !(COLD_RESISTANCE in mutations))
+		var/damage = 0
+		if(breath.temperature <= species.cold_level_1)
+			if(prob(20))
+				to_chat(src, SPAN_DANGER("You feel your face freezing and icicles forming in your lungs!"))
+
+			switch(breath.temperature)
+				if(species.cold_level_3 to species.cold_level_2)
+					damage = COLD_GAS_DAMAGE_LEVEL_3
+				if(species.cold_level_2 to species.cold_level_1)
+					damage = COLD_GAS_DAMAGE_LEVEL_2
+				else
+					damage = COLD_GAS_DAMAGE_LEVEL_1
+
+			apply_damage(damage, BURN, BP_HEAD, used_weapon = "Excessive Cold")
+			fire_alert = 1
```

Comment:
using macros / definitions in this variable helps make the code more understandable by others.

Reply:
Fixed.